tools/SourceKit/tools/sourcekitd/lib/API/sourcekitdAPI-XPC.cpp

```diff
@@ -767,6 +780,13 @@ static sourcekitd_variant_t variantFromXPCObject(xpc_object_t obj) {
     case CustomBufferKind::AttributesArray:
       return {{ (uintptr_t)getVariantFunctionsForAttributesArray(),
                 (uintptr_t)CUSTOM_BUF_START(obj), 0 }};
+    case sourcekitd::CustomBufferKind::SyntaxTree:
+      auto BufStart = CUSTOM_BUF_START(obj);
+      size_t BufSize = *static_cast<const size_t *>(BufStart);
+      return {{ (uintptr_t)getVariantFunctionsForData(),
+                (uintptr_t)BufStart + sizeof(size_t),
+                (uintptr_t)BufSize
+              }};
```

Comment:
Looks like the buffer for `RawData` is supposed to store its size at the beginning. But that seems redundant since `xpc_data_get_length` can get us the total size and just remove the size of the kind value.
E.g. below `#define CUSTOM_BUF_START` add:

```
#define CUSTOM_BUF_SIZE(xobj) xpc_data_get_length(xobj)-sizeof(uint64_t)
```

Then use it as:

```
    case CustomBufferKind::RawData:
      return {{ (uintptr_t)getVariantFunctionsForRawData(),
                (uintptr_t)CUSTOM_BUF_START(obj), CUSTOM_BUF_SIZE(obj) }};
```